frame/dapps-staking/src/tests.rs

```diff
@@ -1289,6 +1289,51 @@ fn claim_after_unregister_is_ok() {
         }
     })
 }
+#[test]
+fn claim_only_payout_is_ok() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+
+        let developer = 1;
+        let staker = 2;
+        let contract_id = MockSmartContract::Evm(H160::repeat_byte(0x01));
+
+        let start_era = DappsStaking::current_era();
+        assert_register(developer, &contract_id);
+        let stake_value = 100;
+        assert_bond_and_stake(staker, &contract_id, stake_value);
+
+        advance_to_era(start_era + 1);
+        assert_ok!(DappsStaking::enable_compound_staking(
+            Origin::signed(staker),
+            RewardHandling::OnlyPayout
+        ));
+
+        assert_claim_staker(staker, &contract_id);
+    })
+}
+
+#[test]
+fn claim_with_zero_staked_is_ok() {
+    ExternalityBuilder::build().execute_with(|| {
+        initialize_first_block();
+
+        let developer = 1;
+        let staker = 2;
+        let contract_id = MockSmartContract::Evm(H160::repeat_byte(0x01));
+
+        let start_era = DappsStaking::current_era();
+        assert_register(developer, &contract_id);
+        let stake_value = 100;
+        assert_bond_and_stake(staker, &contract_id, stake_value);
+
+        advance_to_era(start_era + 1);
+
+        assert_unbond_and_unstake(staker, &contract_id, stake_value);
+
+        assert_claim_staker(staker, &contract_id);
+    })
+}
```

Comment:
How about covering the new extrinsic call with UT?

Also, these scenarios should be covered:
* switch between different reward payouts withing one UT and verify it works
* Produce a scenario where `stakers` vec size would go out of limit
  * verify that if user switches off compounding rewards, claim can once again work